### PR TITLE
User warning: No data available

### PR DESCRIPTION
**Pull request by jadenar07**
## Description of this change
Adds a default to the DataTable, so it displays "There's no data available," whenever a table lacks data
### Before this PR:
image
### After this PR:
Screenshot 2024-07-01 at 12 32 07 PM
## Checklist
- Added a conditional to DataTable.tsx to display the message when the length is 0
- Added the necessary changes to several files to add a Google font

### CODE REVIEW

File: client/index.html [hunk 1]
```diff
@@ -7,6 +7,7 @@
       name="viewport"
       content="width=device-width, initial-scale=1.0, maximum-scale=1"
     />
+    <link href="https://fonts.googleapis.com/css2?family=Roboto+Slab:wght@100..900&display=swap" rel="stylesheet">
```
Comment (mattsahn) [inline client/index.html:10]: What's the rationale for adding this new font download?

File: client/src/pages/DriverLocationInfo/DriverLocationInfo.tsx [hunk 2]
```diff
@@ -30,18 +30,40 @@ import { cn, toastNotify } from "../../utils/ui";
...
   );
-  return drivers.map((curDriver, i) => {
+  /*
+   map() function iterates over an array of drivers and maps each driver to a new array that 
+  includes details about their assigned dropoff locations. For each driver, it 
+  filters the dropoffLocations array to include only those locations where the 
+  driver is scheduled to make deliveries, based on matching location IDs.
+  */
+
+  //Old version of the drivers.map
```
Comment (mattsahn) [inline client/src/pages/DriverLocationInfo/DriverLocationInfo.tsx:54]: No need to keep commented out code.

File: client/src/pages/DriverLocationInfo/DriverLocationInfo.tsx [hunk 3]
```diff
@@ -62,18 +84,26 @@ function processDriversForTable(
...
 //TODO: Make this code cleaner
 function processDropoffLocationsForTable(
   drivers: ProcessedDriver[],
   processedDropOffLocations: ProcessedDropoffLocation[]
 ) {
   return processedDropOffLocations.map((curLocation, i) => {
+    const matchedDrivers = drivers.filter((d) =>
+      d.dropoffLocations.includes(curLocation.id)
+    );
+
```
Comment (mattsahn) [inline client/src/pages/DriverLocationInfo/DriverLocationInfo.tsx:103]: What is this code doing? Doesn't seem related to the overall PR for just adding the "No data available"

Comment (mattsahn) [reply client/src/pages/DriverLocationInfo/DriverLocationInfo.tsx:103]: Oh, I see, it's for below to have count of drivers for the message.

File: client/index.html (hunk 1 above)
Comment (jadenar07) [reply client/index.html:10]: We just thought it would look better for the warning text. Should we take it out?

Comment (mattsahn) [reply client/index.html:10]: Seems like a significant dependency to add solely for the text of an info message that is not used frequently. Let's see what it looks like with the default without it.

**Review by jadenar07 — COMMENTED** [review]
Removed the font additions and commented out code

**Comment by mattsahn** [comment]
> Removed the font additions and commented out code

@jadenar07 Try closing/abandoning this PR and resubmitting to trigger it to deploy in railway
